src/EnergyPlus/EvaporativeFluidCoolers.cc

```diff
@@ -101,7 +101,8 @@ namespace EvaporativeFluidCoolers {
     constexpr std::array<std::string_view, static_cast<int>(EvapLoss::Num)> evapLossNamesUC = {"LOSSFACTOR", "SATURATEDEXIT"};
     constexpr std::array<std::string_view, static_cast<int>(Blowdown::Num)> blowDownNamesUC = {"CONCENTRATIONRATIO", "SCHEDULEDRATE"};
 
-    PlantComponent *EvapFluidCoolerSpecs::factory(EnergyPlusData &state, DataPlant::PlantEquipmentType objectType, std::string const &objectName)
+    EvapFluidCoolerSpecs *
+    EvapFluidCoolerSpecs::factory(EnergyPlusData &state, DataPlant::PlantEquipmentType objectType, std::string const &objectName)
```

Comment:
Again just return the concrete type pointer so that we can call the factory methods and get a meaningful return value rather than just a nebulous `PlantComponent*`.